docs/developer/api/permissions.rst

```diff
@@ -0,0 +1,89 @@
+.. _permissions:
+
+Permission model
+================
+
+Permissions
+-----------
+
+
+Permissions are static, hard-coded identifiers that identify specific actions. Currently, the following permissions
+are defined::
+
+    world:view
+    world:update
+    world:announce
+    world:secrets
+    world:api
+    world:rooms.create
+    room:announce
+    room:view
+    room:update
+    room:delete
+    room:chat.read
+    room:chat.join
+    room:chat.send
```

Comment:
I think so, yes.

* At least having both read and and (join or send) is useful, since it makes sense to allow someone to read but not (join or send), i.e. read-only access.
* Of course it doesn't  make much sense to be able to (join or send), but not read, at least not if we don't abuse chat for things like "anyone can send messages to the speaker but only the speaker can read them".
* Separating join and send into two different permissions allows join (= permanent subscribe = unread notifications) to chat-only rooms you only have read access to.
* Combining join and send into one permission makes handling it a lot trickier on the technical side. It would be weird to just have a "join" permission, since you only join once and if you lose your permission, you should not be able to send any more, so we'd need to check the permission on any send or actively re-check membership status. If it were a "send" permission, it could probably work, we'd need to decide whether we already want to check it when joining.

So yes, I think we could get rid of chat.join, although I think it wouldn't make it less confusing to reason about.

Reply:
I wasn't suggesting to merge `join` and `send`, I'm aware of the usefulness of read-only chats 😉
But yeah, alright, as long as we don't get confused between `join` and `read`, I suppose one more permission check won't impact performance that much – we just have to check `read` permission on every broadcast, since the fact that somebody has `joined` the group doesn't mean that they can also receive messages.

Reply:
Yeah, I hate that we need to check that on every broadcast, but we can probably get away with some caching. Merging join and read wouldn't fix that, though, since you do not need to be joined (only subscribed) to read a channel.